src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -257,6 +261,19 @@ private function compileBinaryOperator(Node\Expr\BinaryOp $node, CompilerContext
         throw new Exception\UnableToCompileNode('Unable to compile binary operator: ' . get_class($node));
     }
 
+    /**
+     * Compile a __DIR__ node
+     *
+     * @param Node\Scalar\MagicConst\Dir $node
+     * @param CompilerContext $context
+     * @return string
+     */
+    private function compileDirConstant(Node\Scalar\MagicConst\Dir $node, CompilerContext $context): string
+    {
+        $fileName = $context->getSelf()->getLocatedSource()->getFileName();
```

Comment:
Try to follow the law of Demeter here; `ReflectionClass` has a method `getFileName()` you can call to avoid using the located class API.

In fact I'd suggest making a new method `getFileName` on the `CompilerContexf` (that calls `ReflectionClass#getFileName`, not the `LocatedSource` API)